lib/galaxy/tool_util/verify/interactor.py

```diff
@@ -981,10 +989,16 @@ def verify_elements(element_objects, element_tests):
 
             element_type = element["element_type"]
             if element_type != "dataset_collection":
+                found_outputs = 1
                 verify_dataset(element, element_attrib, element_outfile)
             else:
                 elements = element["object"]["elements"]
+                found_outputs = len(elements)
                 verify_elements(elements, element_attrib.get("elements", {}))
+            if expected_count is not None and expected_count != found_outputs:
+                raise AssertionError(
+                    f"Element '{element_identifier}': expected to have '{expected_count}' elements, but it had '{found_outputs}'"
```

Comment:
```suggestion
                    f"Element '{element_identifier}': expected to have {expected_count} elements, but it had {found_outputs}"
```
I don't think we need to quote numbers.